crates/project/src/environment.rs

```diff
@@ -194,15 +197,21 @@ async fn load_shell_environment(
 async fn load_shell_environment(
     dir: &Path,
     load_direnv: &DirenvSettings,
-) -> Result<(HashMap<String, String>, Option<DirenvError>)> {
-    use crate::direnv::load_direnv_environment;
+) -> Result<(HashMap<String, String>, Option<EnvironmentErrorMessage>)> {
+    use crate::direnv::{load_direnv_environment, DirenvError};
     use anyhow::{anyhow, Context};
     use std::path::PathBuf;
     use util::parse_env_output;
 
     let (direnv_environment, direnv_error) = match load_direnv {
         DirenvSettings::ShellHook => (None, None),
-        DirenvSettings::Direct => load_direnv_environment(dir).await,
+        DirenvSettings::Direct => match load_direnv_environment(dir).await {
+            Ok(env) => (Some(env), None),
+            Err(err) => (
+                None,
+                <Option<EnvironmentErrorMessage> as From<DirenvError>>::from(err),
+            ),
+        },
```

Comment:
I don't understand why we can't return `Err(err)` here?

Reply:
I don't wanna propagate `DirenvError` any higher than it's necessary, so I've created a common denomination - an `EnvironmentErrorMessage`. We don't return it, because even if there direnv call failed, we can still retrieve information from the shell env

Reply:
Sure, by why not simply return an `Err(anyhow!("")` and then collect `errors.to_string()` further up?